### DISCUSSION TITLE
Dialog does not work outside of component

### DISCUSSION THREAD

**Discussion by haringsrob** [wireui/wireui#556] [Q&A]
Hi,

When I add <x-dialog> to my main template and I have a livewire component being rendered using a regular page blade file the dialog fails with: `Alpine Expression Error: this.$refs.title is undefined`.

However, if I use a full page component, this does work.

Any idea on how I can fix this?

**Comment by haringsrob** [comment]
Realized that it has to be part of the component in order to work because I am not aware of how else it would work.

However, the documentation may need an update to avoid confusion as it uses html tags here: https://livewire-wireui.com/docs/dialogs

**Comment by ph7jack** [comment]
I can't understand the problem.

**Reply by haringsrob** [reply]
I have solved it for now by adding <x-dialog> to my livewire component somewhere.

If I put it higher up in the dom, for example only once after the body tag, it does not work unless I use full page components.

My expectation was that you only have to put the <x-dialog> on the page once.

**Reply by ph7jack** [reply]
Strange behavior. Using or not a full-page livewire the dialog works, because it needs only the alpine.js.
